#**WEEK 1 NONGRADED CHALLENGE**

Kamu adalah salah satu tim relawan back office penanganan COVID-19 di Indonesia. Kamu diminta untuk memperoleh data baru dari dataset yang sudah disediakan. Kerjakan dan jawab pertanyaan-pertanyaan berikut. (**Dataset yang disediakan merupakan jumlah kasus kumulatif tiap harinya**)

Data dapat diakses pada link berikut:
https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/data_covid_dunia_13_apr_22.csv

## **Task 1**

Buatlah sebuah fungsi untuk menghitung penambahan jumlah kasus harian Covid-19 dari data kumulatif yang disediakan! (Hint: Kamu bisa menggunakan **.diff()** yang disediakan oleh Pandas untuk menghitung perbedaan tiap baris data/menghitung penambahan jumlah kasus harian dan perhatikan missing value setelah perhitungan kasus harian, isi missing value dengan angka 0!)

In [80]:
import numpy as np
import pandas as pd

In [81]:
df = pd.read_csv("data_covid_dunia_13_apr_22.csv")

In [82]:
df

Unnamed: 0  Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
0      1/22/20            0        0        0        0       0           0   
1      1/23/20            0        0        0        0       0           0   
2      1/24/20            0        0        0        0       0           0   
3      1/25/20            0        0        0        0       0           0   
4      1/26/20            0        0        0        0       0           0   
..         ...          ...      ...      ...      ...     ...         ...   
808     4/9/22       177974   274191   265720    40328   99194          11   
809    4/10/22       177974   274219   265724    40328   99194          11   
810    4/11/22       178141   274219   265727    40328   99194          11   
811    4/12/22       178257   274272   265730    40328   99194          11   
812    4/13/22       178295   274320   265731    40709   99194          11   

     Antigua and Barbuda  Argentina  Armenia  ...  Uruguay  Uzbekistan  \
0                      0          0        0  ...        0           0   
1                      0          0        0  ...        0           0   
2                      0          0        0  ...        0           0   
3                      0          0        0  ...        0           0   
4                      0          0        0  ...        0           0   
..                   ...        ...      ...  ...      ...         ...   
808                 7511    9052083   422662  ...   893575      238040   
809                 7511    9052536   422677  ...   893909      238071   
810                 7511    9054126   422678  ...   894267      238088   
811                 7523    9056203   422691  ...   894630      238098   
812                 7523    9057923   422711  ...   894984      238145   

     Vanuatu  Venezuela   Vietnam  West Bank and Gaza  Winter Olympics 2022  \
0          0          0         0                   0                     0   
1          0          0         2                   0                     0   
2          0          0         2                   0                     0   
3          0          0         2                   0                     0   
4          0          0         2                   0                     0   
..       ...        ...       ...                 ...                   ...   
808     5125     521373  10169929              656287                   535   
809     5480     521471  10198236              656287                   535   
810     5487     521471  10250160              656287                   535   
811     5536     521546  10272964              656617                   535   
812     5543     521714  10297587              656617                   535   

     Yemen  Zambia  Zimbabwe  
0        0       0         0  
1        0       0         0  
2        0       0         0  
3        0       0         0  
4        0       0         0  
..     ...     ...       ...  
808  11813  317483    246925  
809  11813  317483    246958  
810  11814  317804    247010  
811  11814  317804    247094  
812  11815  318113    247160  

[813 rows x 285 columns]

In [83]:
df.rename(columns={"Unnamed: 0": "Tanggal"}, inplace=True)

In [84]:
df.set_index("Tanggal", inplace= True)

In [85]:
df.diff().fillna(0)

Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
Tanggal                                                               
1/22/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/23/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/24/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/25/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/26/20          0.0      0.0      0.0      0.0     0.0         0.0   
...              ...      ...      ...      ...     ...         ...   
4/9/22           0.0     55.0      6.0      0.0     0.0         0.0   
4/10/22          0.0     28.0      4.0      0.0     0.0         0.0   
4/11/22        167.0      0.0      3.0      0.0     0.0         0.0   
4/12/22        116.0     53.0      3.0      0.0     0.0         0.0   
4/13/22         38.0     48.0      1.0    381.0     0.0         0.0   

         Antigua and Barbuda  Argentina  Armenia  Australia  ...  Uruguay  \
Tanggal                                                      ...            
1/22/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/23/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/24/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/25/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/26/20                  0.0        0.0      0.0        0.0  ...      0.0   
...                      ...        ...      ...        ...  ...      ...   
4/9/22                   0.0      840.0     19.0      785.0  ...    566.0   
4/10/22                  0.0      453.0     15.0      747.0  ...    334.0   
4/11/22                  0.0     1590.0      1.0      909.0  ...    358.0   
4/12/22                 12.0     2077.0     13.0      967.0  ...    363.0   
4/13/22                  0.0     1720.0     20.0     1057.0  ...    354.0   

         Uzbekistan  Vanuatu  Venezuela  Vietnam  West Bank and Gaza  \
Tanggal                                                                
1/22/20         0.0      0.0        0.0      0.0                 0.0   
1/23/20         0.0      0.0        0.0      2.0                 0.0   
1/24/20         0.0      0.0        0.0      0.0                 0.0   
1/25/20         0.0      0.0        0.0      0.0                 0.0   
1/26/20         0.0      0.0        0.0      0.0                 0.0   
...             ...      ...        ...      ...                 ...   
4/9/22         46.0    203.0       48.0  34140.0                 0.0   
4/10/22        31.0    355.0       98.0  28307.0                 0.0   
4/11/22        17.0      7.0        0.0  51924.0                 0.0   
4/12/22        10.0     49.0       75.0  22804.0               330.0   
4/13/22        47.0      7.0      168.0  24623.0                 0.0   

         Winter Olympics 2022  Yemen  Zambia  Zimbabwe  
Tanggal                                                 
1/22/20                   0.0    0.0     0.0       0.0  
1/23/20                   0.0    0.0     0.0       0.0  
1/24/20                   0.0    0.0     0.0       0.0  
1/25/20                   0.0    0.0     0.0       0.0  
1/26/20                   0.0    0.0     0.0       0.0  
...                       ...    ...     ...       ...  
4/9/22                    0.0    1.0     0.0       0.0  
4/10/22                   0.0    0.0     0.0      33.0  
4/11/22                   0.0    1.0   321.0      52.0  
4/12/22                   0.0    0.0     0.0      84.0  
4/13/22                   0.0    1.0   309.0      66.0  

[813 rows x 284 columns]

## **Task 2**

Buat variabel baru yang berisikan data penambahan jumlah kasus positif harian Covid-19 di dunia menggunakan fungsi yang telah kamu buat sebelumnya!


In [86]:
df_harian = df.diff().fillna(0)

## **Task 3**

Pada tanggal berapa puncak tertinggi kasus harian Covid-19 di Indonesia? bandingkan dengan tanggal 10 April 2022, apakah mengalami penurunan atau kenaikan jumlah kasus dibandingkan tanggal tersebut? (Kamu bisa gunakan pandas query untuk hal ini)


In [87]:
df_harian["Indonesia"].sort_values(ascending=False)

Tanggal
2/16/22    64718.0
2/17/22    63956.0
2/23/22    61488.0
2/18/22    59635.0
2/19/22    59384.0
            ...   
2/16/20        0.0
2/15/20        0.0
2/14/20        0.0
2/13/20        0.0
1/22/20        0.0
Name: Indonesia, Length: 813, dtype: float64

In [88]:
df_harian.loc["4/10/22","Indonesia"]

1071.0

Diketahui puncak kasus penambahan covid 19 terjadi pada tanggal 16 Februari 2022 dengan jumlah kasus 64718.
Sementara itu pada tanggal 10 April 2022 penambahan kasus covid 19 turun ke angka 1071 kasus

## **Task 4**

Gunakan for loop dan conditional if untuk membuat sebuah list yang berisikan penambahan jumlah kasus harian Covid-19 dengan ketentuan hanya mengambil data Indonesia saja dan jumlah kasus di bawah 10000. Berapa jumlah hari dengan kasus di bawah 10000? (Gunakan for loop dan conditional if dalam hal ini)

In [89]:
df_harian.reset_index(drop=True, inplace=True)

In [90]:
df_harian.shape

(813, 284)

In [91]:
jumlah = []

for i in range(0, len(df_harian)):
    if df_harian.loc[i, "Indonesia"] < 10000:
        jumlah.append(df_harian.loc[i, "Indonesia"])
    elif df_harian.loc[i, "Indonesia"] > 10000:
        continue

jumlah


[0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 0.0,
 2.0,
 0.0,
 0.0,
 0.0,
 2.0,
 0.0,
 2.0,
 13.0,
 8.0,
 7.0,
 0.0,
 35.0,
 27.0,
 21.0,
 17.0,
 38.0,
 55.0,
 84.0,
 58.0,
 81.0,
 64.0,
 65.0,
 107.0,
 104.0,
 103.0,
 153.0,
 109.0,
 130.0,
 129.0,
 114.0,
 149.0,
 113.0,
 196.0,
 106.0,
 181.0,
 218.0,
 247.0,
 218.0,
 337.0,
 219.0,
 330.0,
 399.0,
 316.0,
 282.0,
 297.0,
 380.0,
 407.0,
 325.0,
 327.0,
 185.0,
 375.0,
 283.0,
 357.0,
 436.0,
 396.0,
 275.0,
 214.0,
 415.0,
 260.0,
 347.0,
 433.0,
 292.0,
 349.0,
 395.0,
 484.0,
 367.0,
 338.0,
 336.0,
 533.0,
 387.0,
 233.0,
 484.0,
 689.0,
 568.0,
 490.0,
 529.0,
 489.0,
 496.0,
 486.0,
 693.0,
 973.0,
 634.0,
 949.0,
 526.0,
 479.0,
 415.0,
 686.0,
 687.0,
 678.0,
 557.0,
 700.0,
 467.0,
 609.0,
 684.0,
 585.0,
 703.0,
 993.0,
 672.0,
 847.0,


In [93]:
len(jumlah)

661

Jumlah penambahan kasus harian pada dataframe yang berjumlah kurang dari 10000 kasus diketahui tercatat sebanyak 661 hari